rosetta-source/src/main/rosetta/product-qualification-func.rosetta

```diff
@@ -27,168 +57,154 @@ func Qualify_AssetClass_InterestRate: <"Qualifies a product as having the Asset
-    alias forwardUnderlier:
-        economicTerms -> payout -> forwardPayout only-element -> underlier
+    alias optionUnderlier: economicTerms -> payout -> optionPayout only-element -> optionUnderlier
+    alias settlementUnderlier:
+        economicTerms -> payout -> settlementPayout only-element -> underlier
 
     set is_product:
-        (economicTerms -> payout -> performancePayout -> underlier
-                    extract [ Qualify_UnderlierProduct_Equity(item) ]
-                all = True
-                and ( // Interest Rate Payout + Performance Payout (Price Return Swap, Total Return Swap)
-                (economicTerms -> payout -> interestRatePayout, economicTerms -> payout -> performancePayout) only exists
-                    // Performance Payout + Fixed Price Payout (Dividend Swap)
-                    or (economicTerms -> payout -> performancePayout, economicTerms -> payout -> fixedPricePayout) only exists
-                        // Performance Payout only (Variance, Volatility and Correlation Swap)
-                    or economicTerms -> payout -> performancePayout only exists))
+        (economicTerms -> payout -> performancePayout -> underlier -> Observable
+                extract Qualify_UnderlierObservable_Equity
+                then all = True
+            and ( // Interest Rate Payout + Performance Payout (Price Return Swap, Total Return Swap)
+            (economicTerms -> payout -> interestRatePayout, economicTerms -> payout -> performancePayout) only exists
+                // Performance Payout + Fixed Price Payout (Dividend Swap)
+                or (economicTerms -> payout -> performancePayout, economicTerms -> payout -> fixedPricePayout) only exists
+                    // Performance Payout only (Variance, Volatility and Correlation Swap)
+                or economicTerms -> payout -> performancePayout only exists))
             or (economicTerms -> payout -> optionPayout only exists
-                and if optionUnderlier exists
-                    then (Qualify_UnderlierProduct_Equity(optionUnderlier) = True
-                            or optionUnderlier -> index ->> assetClass = AssetClassEnum -> Equity
-                            or optionUnderlier -> security -> productTaxonomy -> primaryAssetClass any = AssetClassEnum -> Equity)
-                    else False)
-            or (economicTerms -> payout -> forwardPayout only exists
-                and if forwardUnderlier exists
-                    then (Qualify_UnderlierProduct_Equity(forwardUnderlier)
-                            or forwardUnderlier -> index ->> assetClass = AssetClassEnum -> Equity
-                            or forwardUnderlier -> security -> productTaxonomy -> primaryAssetClass any = AssetClassEnum -> Equity)
-                    else False)
+                and (ObservableQualification(optionUnderlier -> Observable, SecurityTypeEnum -> Equity, AssetClassEnum -> Equity) 
```

Comment:
Should this not use `Qualify_UnderlierObservable_Equity`, which casts a wider net than just `SecurityTypeEnum -> Equity`? This is what the original logic uses.

Same comment below for the settlement payout case.

Reply:
Check logic

Reply:
updated both instances